packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -43,25 +43,45 @@ import QuickAccess from '../quick-access';
  * and contains the "restricted" application part.
  */
 
+const getHasUnauthorizedError = graphQLErrors =>
+  graphQLErrors.find(
+    gqlError =>
+      gqlError.extensions &&
+      gqlError.extensions.code &&
+      gqlError.extensions.code === 'UNAUTHENTICATED'
+  );
+const getHasUserBeenDeletedError = graphQLErrors =>
+  graphQLErrors.find(
+    gqlError =>
+      gqlError.extensions &&
+      gqlError.extensions.message &&
+      // NOTE: The CTP API does not provide an error code in this case.
+      gqlError.extensions.message.includes('was not found.')
```

Comment:
I don't really like this but CTP does not provide a better error yet. I also don't want to wrap CTP's API in our gateway as we don't do that in other cases (e.g. when loading an org). I think we should push CTP to provide better error codes and then come back here.